GestureHandler.web.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import {
+  ScrollView,
+  Slider,
+  Switch,
+  TextInput,
+  ToolbarAndroid,
+  ViewPagerAndroid,
+  DrawerLayoutAndroid,
+  FlatList,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import gestureHandlerRootHOC from './gestureHandlerRootHOC';
+import Directions from './Directions';
+import State from './State';
+
+// Factory for stub Handler components
+function createStubHandler(name) {
+  return class NativeViewGestureHandler extends React.Component {
+    static displayName = name;
+
+    container = React.createRef();
+
+    setNativeProps() {
+      // Since this is a stub we do not need to pass on native props.
+      // However, we need to implement it here to avoid null calls.
+      // If for any reason we need to pass on native props we can fallback to this:
+      // this.container.current.setNativeProps(...args)
+    }
+
+    render() {
+      const { children, ...rest } = this.props;
+
+      // We don't want to create another layer, so instead we clone it only but keep the reference
+      const child = React.Children.only(children);
+      return React.cloneElement(child, {
+        ref: this.container,
+        ...rest,
+      });
+    }
+  };
+}
+
+// Create all Handler components with their respective handler functions
+// (at the moment only TapGestureHandler is properly supported)
+const NativeViewGestureHandler = createStubHandler('NativeViewGestureHandler');
+
+class TapGestureHandler extends React.Component {
+  setNativeProps() {}
+
+  render() {
+    const { children, enabled, onHandlerStateChange, style } = this.props;
+
+    return (
+      <TouchableWithoutFeedback
+        style={style}
+        onPress={({ x }) => {
```

Comment:
Hm, can't we make it more complex? I believe it's fully possible to achieve more similar behaviour to native `TapGestureHandler` by by combing `onPressIt`, `onPressOut` and `pressRetentionOffset`. Is it possible on web?

Reply:
Yeah, I think in the future we should try to respect the gesture props even more. `pressRetentionOffset` means something else if I understand correctly and cannot be mapped to any of the props. As a start I implemented support for `maxDurationMs` prop but I think additional props should be part of future PRs. What do you think?